kiwi/solverimpl.h

```diff
@@ -138,7 +142,7 @@ class SolverImpl
 	*/
 	void removeConstraint( const Constraint& constraint )
 	{
-		CnMap::iterator cn_it = m_cns.find( constraint );
+		auto cn_it = m_cns.find( constraint );
```

Comment:
Using auto in this case and the followings does not bring anything and I would argue even reduce the code readability so I would prefer to keep the explicit notation.

Reply:
This code is just more idiomatic, no one writing C++11 uses explicit type for iterator.

But yeah, if you insist I can rollback this.

Reply:
Yes please, I touch the c++ part in a fairly infrequent manner so less magic makes it easier. As a said I am fine with the changes in the loop and understand the use of auto there though.

Reply:
No problem, I'll take care of that tomorrow, as it's past midnight in France. 🙂

Reply:
Sure no hurry.

Reply:
Actually I'm with @SirLynx here - getting the value of `find` is usually stored in an `auto` because it's implied the type is an iterator belonging to the collection's type - which is otherwise very verbose to write and read.

The use of `auto` in this PR isn't what Is call egregious.

Reply:
I feel in minority now... I guess I will be able to remember that part for iterators. You can revert the commit and I will merge.

Reply:
Thank you for reconsidering this, I reverted my commit.